pkg-manager/plugin-commands-installation/src/recursive.ts

```diff
@@ -139,8 +140,9 @@ export async function recursive (
     linkWorkspacePackagesDepth: opts.linkWorkspacePackages === 'deep' ? Infinity : opts.linkWorkspacePackages ? 0 : -1,
     ownLifecycleHooksStdio: 'pipe',
     peer: opts.savePeer,
-    pruneLockfileImporters: ((opts.ignoredPackages == null) || opts.ignoredPackages.size === 0) &&
-      pkgs.length === allProjects.length,
+    pruneLockfileImporters: opts.pruneLockfileImporters ??
```

Comment:
One other option: We could name the publicly exposed option `produceFilteredWantedLockfile` and have that value imply the internal `pruneLockfileImporters` value here.

The idea of `produceFilteredWantedLockfile` would be to intentionally create a wanted lockfile that omits certain importers, which is the goal of `pnpm deploy`. There could probably be a better name for `produceFilteredWantedLockfile` if we like that idea.